luda-editor/luda-editor-client/src/app/types/clip.rs

```diff
@@ -0,0 +1,36 @@
+use std::collections::HashMap;
+
+use namui::Language;
+use serde::{Deserialize, Serialize};
+
+use super::{CameraAngle, Time};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CameraClip {
+    pub id: String,
+    pub start_at: Time,
+    pub end_at: Time,
+    pub camera_angle: CameraAngle,
+}
+
+pub enum Clip<'a> {
+    Camera(&'a CameraClip),
+    Subtitle(&'a SubtitleClip),
+}
+pub enum MutableClip<'a> {
+    Camera(&'a mut CameraClip),
+    Subtitle(&'a mut SubtitleClip),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SubtitleClip {
+    pub id: String,
+    pub start_at: Time,
+    pub subtitle: Subtitle,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Subtitle {
```

Comment:
I have no idea that it is the good to put Subtitle struct in here. You moved structs to each specific files from mod.rs, some structs went to very specific file but some structs (like Subtitle or Circumscribed) just left in file which contains that struct. 
Is Subtitle struct related to clip? Is there dependency between clip and Subtitle? Clip needs Subtitles but Subtitles doesn't.  If I make a code which uses Subtitle, but not clip, I should import clip.rs because clip has subtitle even if I don't use clip. I think it's weird. 

I suggest making common file to put types about editor like editor_common.rs or make each specific file like subtitle.rs and circumscribed.rs. Personally I think the first one is better than seconds because this PR is about router, not a type files, and first idea keep the same type file system.

Reply:
> I should import clip.rs because clip has subtitle even if I don't use clip.

Actually, thanks to re-export in mod.rs, we can use it with just via importing types like `App::types::Subtitle`.
But yes, `Subtitle` and `Circumscribed` are in wrong place now.

> Personally I think the first one is better than seconds because this PR is about router, not a type files,

I agree. I will rollback and do it later


c698ce0